builder/simplify.cc

```diff
@@ -470,6 +524,8 @@ class simplifier : public node_mutator {
     } else {
       set_result(loop::make(op->sym, op->mode, std::move(bounds), std::move(step), std::move(body)));
     }
+
+    merge_loop_references(std::move(old_references));
```

Comment:
This call should set the new parameter to true (since it is in fact still a loop).